docs/source/1.0/spec/core/behavior-traits.rst

```diff
@@ -655,3 +655,134 @@ See
+    for the operation's HTTP Request.
+
+Trait selector
+    ``operation``
+Value type
+    ``structure``
+
+The `httpChecksum` trait is a structure that contains the following members:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
+
+    * - Property
+      - Type
+      - Description
+    * - request
+      - :ref:`HttpChecksumProperties structure <checksum-properties>`
+      - The `request` structure values define checksum validation behavior for
+        HTTP Request.
+
+    * - response
+      - :ref:`HttpChecksumProperties structure <checksum-properties>`
+      - The `response` structure values define checksum validation behavior for
+        HTTP Response.
+
+
+.. _checksum-properties:
+
+HttpChecksumProperties structure
+================================
+
+HttpChecksumProperties defines checksum validation behavior using the
+following members:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
+
+    * - Property
+      - Type
+      - Description
+    * - prefix
+      - ``string``
+      - **Required**. The prefix string is used to construct a header for a
+        checksum type. Prefix is a required member, and SHOULD be lower-cased.
+        The recommended ABNF to follow for prefix is `([a-z]*([a-z]/[0-9]/"-")*`.
+
+        To construct a header name for usage with checksum, follow the pattern:
+        `prefix + algorithm-name`. For example, if prefix is "x-checksum-"
+        and "sha256" algorithm is used for computing checksum, header name
+        would resolve as "x-checksum-sha256".
+
+    * - location
+      - ``string``
+      - A string representing checksum location. The string value MUST be a
+        valid :ref:`Location enum <checksum-location-enum>`. By default,
+        location resolves to "header" value. Service MUST always support
+        checksum in header for both requests and responses.
+
+        Optionally, for request, location can be modeled as "trailer"
+        indicating service supports checksum in trailer, in addition to
+        support in header.
+
+    * - algorithms
+      - ``[string]``
+      - **Required**. List of strings representing checksum algorithms
+        supported for the HTTP Request or Response. Algorithms is a
+        required member, and a valid algorithm name SHOULD follow
+        ABNF: `([a-z]* [0-9]*)*`. Algorithm name SHOULD be lower-cased
+        and not hyphenated.
+
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        @httpChecksum(
+            request: {
+                location: "trailer",
+                prefix: "x-checksum-",
+                algorithms: ["sha256", "crc32"]
+            },
+            response: {
+                prefix: "x-checksum-",
+                algorithms: ["sha256", "crc32"]
+            }
+        )
+        operation PutSomething {
+            input: PutSomethingInput,
+            output: PutSomethingOutput
+        }
+
+
+.. _checksum-location-enum:
```

Comment:
This is small enough where it doesn't need its own section, it can just be part of the property explanation (like [the `error` trait value type](https://awslabs.github.io/smithy/1.0/spec/core/type-refinement-traits.html#error-trait).)